sdk/communication/Azure.Communication.JobRouter/README.md

```diff
@@ -0,0 +1,298 @@
+
+### Exception policy
+An exception policy controls the behavior of a Job based on a trigger and executes a desired action. The exception policy is attached to a Queue so it can control the behavior of Jobs in the Queue.
+
+## Examples
+
+### Distribution Policy
+Before we can create a Queue, we need a Distribution Policy.
+
+```C# Snippet:Azure_Communication_JobRouter_Tests_Samples_CreateDistributionPolicyLongestIdleTTL1D_Async
+var distributionPolicy = await routerAdministrationClient.CreateDistributionPolicyAsync(
+    new CreateDistributionPolicyOptions(
+        distributionPolicyId: "distribution-policy-1",
+        offerTtl: TimeSpan.FromDays(1),
+        mode: new LongestIdleMode())
+);
+```
+
+### Queue
+Next, we can create the queue.
+```C# Snippet:Azure_Communication_JobRouter_Tests_Samples_CreateQueue_Async
+var queue = await routerAdministrationClient.CreateQueueAsync(
+    new CreateQueueOptions(
+        queueId: "queue-1",
+        distributionPolicyId: distributionPolicy.Value.Id)
+);
+```
+
+### Job
+Now, we can submit a job directly to that queue, with a worker selector the requires the worker to have the label `Some-Skill` greater than 10.
+```C# Snippet:Azure_Communication_JobRouter_Tests_Samples_CreateJobDirectQAssign_Async
+var job = await routerClient.CreateJobAsync(
+    new CreateJobOptions(
+        jobId: "jobId-1",
+        channelId: "my-channel",
+        queueId: queue.Value.Id)
+    {
+        ChannelReference = "12345",
+        Priority = 1,
+        RequestedWorkerSelectors = new List<WorkerSelector>
+        {
+            new WorkerSelector("Some-Skill", LabelOperator.GreaterThan, new LabelValue(10))
+        }
+    });
+```
+
+### Worker
+Now, we register a worker to receive work from that queue, with a label of `Some-Skill` equal to 11.
+```C# Snippet:Azure_Communication_JobRouter_Tests_Samples_RegisterWorker_Async
+var worker = await routerClient.CreateWorkerAsync(
+    new CreateWorkerOptions(
+        workerId: "worker-1",
+        totalCapacity: 1)
+    {
+        QueueIds = new Dictionary<string, QueueAssignment>() { [queue.Value.Id] = new QueueAssignment() },
+        Labels = new Dictionary<string, LabelValue>()
+        {
+            ["Some-Skill"] = new LabelValue(11)
+        },
+        ChannelConfigurations = new Dictionary<string, ChannelConfiguration>()
+        {
+            ["my-channel"] = new ChannelConfiguration(1)
+        },
+        AvailableForOffers = true,
+    }
+);
+```
+
+### Offer
+We should get a [RouterWorkerOfferIssued][offer_issued_event_schema] from our [EventGrid subscription][subscribe_events].
+
+There are several different Azure services that act as a [event handler][event_grid_event_handlers].
+For this scenario, we are going to assume Webhooks for event delivery. [Learn more about Webhook event delivery][webhook_event_grid_event_delivery]
+
+Once events are delivered to the event handler, we can deserialize the JSON payload into a list of events.
+
+```C#
```

Comment:
These samples should be injected via snippet.

Reply:
This particular sample is a duplicate of https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/eventgrid/Azure.Messaging.EventGrid#receiving-and-deserializing-events

In this case, we are specifying that this is _one of the way_ to consume the event from EventHub. If this snippet is to be injected, we'll need to take a dependency on event grid sdk.

Reply:
Your tests should either take that dependency or reference the Event Grid snippet directly here.   In the latter case, I'd suggest adding a note in the Event Grid source to note the snippet is used outside that scope.

The overall goal is to ensure that all code that appears in .NET SDK samples is compiled/tested and cannot rot over time.

Reply:
This has been updated to be injected with a snippet